client/src/pages/ShoppingCartPage.js

```diff
@@ -1,18 +1,173 @@
-import React from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import React, { useState } from "react";
 import { Container, Row } from "react-bootstrap";
-import { NavbarComponent } from "../ui-components/NavbarComponent/NavbarComponent";
+import { Redirect } from "react-router-dom";
 
-import "bootstrap/dist/css/bootstrap.min.css";
+import { NavbarComponent } from "../ui-components/NavbarComponent/NavbarComponent";
+import { ShoppingCartTitle } from "../ui-components/ShoppingCartComponent/ShoppingCartTitle";
+import { ShoppingCartTable } from "../ui-components/ShoppingCartComponent/ShoppingCartTable";
+import { ShoppingCartTotAmount } from "../ui-components/ShoppingCartComponent/ShoppingCartTotAmount";
+import { ShoppingCartControls } from "../ui-components/ShoppingCartComponent/ShoppingCartControls";
+import { ModalOrderConfirmed } from "../ui-components/ShoppingCartComponent/ModalOrderConfirmed";
 
+import { createOrder } from "../services/ApiClient";
 function ShoppingCartPage(props) {
+  // as props, ShoppingCartPage receives
+  //      - a Map <ItemID, Qty>
+  //      - the client
+  // it mantains as state
+  //      - a cart (Map <ItemID, Qty>)
+  //      - the total amount of the current cart
+  // it uses function getProductById(id) -> product object
+  // it uses function getClientById(id) -> client object
+
+  /* MOCK DATA (Map, client, getProductById) */
+  const propsMap = new Map();
+  propsMap.set(1, 1);
+  propsMap.set(2, 1);
+  propsMap.set(3, 1);
+  propsMap.set(4, 1);
+
+  // This is the format for the backend, I am not sure if can we use this format????
+  // Mock data, same with the postman request
+  // If we can get this format from props, then we are totally done!
+
+  const mockProductQuantity = [
+    { productId: "718d971d89d6240eb03742d7", quantity: 1 },
+    { productId: "298d971d89d6240eb03742d7", quantity: 1 },
+    { productId: "318d971d89d6240eb03742d7", quantity: 1 },
+    { productId: "418d971d89d6240eb03742d7", quantity: 1 },
+  ];
+
+  const propsClientId = "918d971d89d6240eb03742d7";
+
+  const getProductById = (id) => {
+    const staticProducts = [
+      {
+        id: "718d971d89d6240eb03742d7",
+        item: "Eggplant",
+        description: "Origin: Italy, packaging: 1kg",
+        price: 2.5,
+      },
+      {
+        id: "298d971d89d6240eb03742d7",
+        item: "Zucchini",
+        description: "Origin: Italy, packaging: 1kg",
+        price: 3,
+      },
+      {
+        id: "318d971d89d6240eb03742d7",
+        item: "Eggs",
+        description: "Origin: Italy, packaging: 6 eggs",
+        price: 1.6,
+      },
+      {
+        id: "418d971d89d6240eb03742d7",
+        item: "Milk",
+        description: "Origin: Italy, packaging: 2l",
+        price: 1,
+      },
+    ];
+    return staticProducts.filter((p) => p.id === id)[0];
+  };
```

Comment:
In the last PR, getProductById was added, try to use it instead of mock data

Reply:
to be more specific: to get the products you have to make a get ".../products?ids="+list of all ids divided by a comma

see GetProductsByIDs.http for more examples